src/Compilers/CSharp/Test/Semantic/SourceGeneration/SyntaxAwareGeneratorTests.cs

```diff
@@ -192,21 +193,174 @@ void Function()
             Assert.IsType<CompilationUnitSyntax>(testReceiver.VisitedNodes[0]);
         }
 
+        [Fact]
+        public void Syntax_Receiver_Exception_During_Creation()
+        {
+            var source = @"
+class C 
+{
+    int Property { get; set; }
+
+    void Function()
+    {
+        var x = 5;
+        x += 4;
+    }
+}
+";
+            var parseOptions = TestOptions.Regular;
+            Compilation compilation = CreateCompilation(source, options: TestOptions.DebugDll, parseOptions: parseOptions);
+            compilation.VerifyDiagnostics();
+
+            Assert.Single(compilation.SyntaxTrees);
+
+            Exception ex = new Exception("Test Exception");
+
+            var testGenerator = new CallbackGenerator(
+                onInit: (i) => i.RegisterForSyntaxNotifications(() => throw ex),
+                onExecute: (e) => { }
+                );
+
+            GeneratorDriver driver = new CSharpGeneratorDriver(parseOptions, ImmutableArray.Create<ISourceGenerator>(testGenerator), CompilerAnalyzerConfigOptionsProvider.Empty, ImmutableArray<AdditionalText>.Empty);
+            driver = driver.RunFullGeneration(compilation, out var outputCompilation, out var outputDiagnostics);
+            var results = driver.GetRunResult();
+
+            Assert.Empty(results.SyntaxTrees);
+            Assert.Single(results.Diagnostics);
+            Assert.Single(results.Results);
+            Assert.Single(results.Results[0].Diagnostics);
+
+            Assert.NotNull(results.Results[0].Exception);
+            Assert.Equal("Test Exception", ex.Message);
```

Comment:
>ex [](start = 43, length = 2)

We're testing the local rather than the exception from `results`. Same comment for other tests.